mockk-native/src/main/kotlin/io/mockk/impl/instantiation/NativeMockFactory.kt

```diff
@@ -0,0 +1,35 @@
+package io.mockk.impl.instantiation
+
+import io.mockk.MethodDescription
+import io.mockk.impl.log.Logger
+import io.mockk.impl.stub.Stub
+import io.mockk.impl.stub.StubGatewayAccess
+import io.mockk.impl.stub.StubRepository
+import kotlin.reflect.KClass
+
+class NativeMockFactory(
+    stubRepository: StubRepository,
+    instantiator: NativeInstantiator,
+    gatewayAccess: StubGatewayAccess
+) :
+    AbstractMockFactory(
+        stubRepository,
+        instantiator,
+        gatewayAccess
+    ) {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : Any> newProxy(
+        cls: KClass<out T>,
+        moreInterfaces: Array<out KClass<*>>,
+        stub: Stub,
+        useDefaultConstructor: Boolean,
+        instantiate: Boolean
+    ): T {
+        TODO("proxy")
```

Comment:
Proxies would be even harder to implement in native world than new by KClass. Could it be avoided?

Reply:
Proxy = empty object + interception. I am experimenting with IR transformation right now for interception. The problem I see is that libraries compiled earlier wont be transformed, so need special distribution of transformed libraries